internal/users/users_edit.go

```diff
@@ -0,0 +1,145 @@
+package users
+
+import (
+	"Twitter_like_application/internal/database/pg"
+	"Twitter_like_application/internal/services"
+	"encoding/json"
+	"fmt"
+	"gopkg.in/go-playground/validator.v9"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+func EditProfile(w http.ResponseWriter, r *http.Request) {
+	userValid := &UserValid{
+		validate: validator.New(),
+		validErr: make(map[string]string),
+	}
+
+	err := userValid.validate.RegisterValidation("checkPassword", func(fl validator.FieldLevel) bool {
+		return CheckPassword(fl, userValid)
```

Comment:
both `RegisterValidation` and all the check can be methods of `UserValid` type, it would simplify things and make it reusable.